### PR TITLE
Very fast vertica data loading

### LINKED ISSUE #14472: Speed up test data loading for Vertica (etc)
Tests take _forever_ because loading the sample dataset in Vertica and syncing literally takes over 8 minutes in CircleCI.

Sync in general is slow:

```
Sync :vertica Database sad-toucan-incidents took 28.4 s
load-data for vertica toucan-microsecond-incidents incidents took 47.0 ms
Sync :vertica Database toucan-microsecond-incidents took 26.9 s
```

I'm not sure why it takes 30 seconds to sync a database with a hundred rows in a single table -- that's an open question -- but maybe by skipping fingerprinting or some of the other more advanced sync steps (by disabling them in the Database properties) we can speed things up a bit.

Recently it took almost _half an hour_ to run Vertica tests on CI

### PR DESCRIPTION
**Pull request by camsaul**
Fixes #14472

### BEFORE:

```
load-data for vertica sample-dataset products took 2.6 s
load-data for vertica sample-dataset people took 36.6 s
load-data for vertica sample-dataset reviews took 11.8 s
load-data for vertica sample-dataset orders took 3.3 mins
```

### AFTER

```
load-data for vertica sample-dataset products took 48.7 ms
load-data for vertica sample-dataset people took 114.0 ms
load-data for vertica sample-dataset reviews took 75.0 ms
load-data for vertica sample-dataset orders took 239.3 ms
```

A few other random formatting fixes are in here as well

### CODE REVIEW

File: src/metabase/util.clj [hunk 1]
```diff
@@ -795,18 +795,45 @@
...
-      (prog1 (do ~@body)
-        (println (format-color '~'green "%s took %s"
-                   ~message
-                   (format-nanoseconds (- (System/nanoTime) start-time#))))))))
+   `(let [message#    ~message
+          start-time# (System/nanoTime)
+          result#     (binding [*profile-level* (inc *profile-level*)]
+                        ~@body)]
+      (profile-print-time message# start-time#)
+      result#)))
```
Comment (camsaul) [inline src/metabase/util.clj:836]: I was profiling stuff to try to figure out why this is so slow and I made this change so nested calls to `u/profile` are indented like such:

```
   ↙ sync-db-metadata! took 4.2 s
     ↙ fingerprint-fields-for-db! took 14.1 s
     ↙ classify-fields-for-db! took 174.5 ms
     ↙ classify-tables-for-db! took 24.1 ms
   ↙ analyze-db! took 14.3 s
   ↙ update-field-values! took 1.1 s
 ↙ sync database took 19.7 s
Sync :vertica Database test-data took 20.6 
```

Not specific to this but I think it's a good improvement we should keep

File: test/metabase/api/dataset_test.clj [hunk 2]
```diff
@@ -47,8 +47,10 @@
        :else
        [k v]))))
 
-(defn- most-recent-query-execution []
-  (db/select-one QueryExecution {:order-by [[:started_at :desc]]}))
+(defn- most-recent-query-execution-for-query [query]
+  (db/select-one QueryExecution
+    :hash (qp-util/query-hash query)
+    {:order-by [[:started_at :desc]]}))
```
Comment (camsaul) [inline test/metabase/api/dataset_test.clj:53]: This should fix random test failures we were seeing on occasion

File: modules/drivers/vertica/test/metabase/test/data/vertica.clj [hunk 3]
```diff
@@ -54,28 +59,94 @@
...
+            (when-not (= actual-num-rows (count rows))
+              (throw (ex-info (format "Expected count(*) to return %d, but only got" (count rows) actual-num-rows)
+                              {:inserted-rows (take 100 (actual-rows))}))))
+          ;; success!
+          :ok
+          (catch Throwable e
+            (throw (ex-info "Error loading rows from CSV file"
+                            {:filename filename
+                             :rows     (take 10 (str/split-lines (slurp filename)))}
+                            e))))))))
```
Comment (robdaemon) [inline modules/drivers/vertica/test/metabase/test/data/vertica.clj:140]: This stuff might be useful elsewhere, like Redshift. https://docs.aws.amazon.com/redshift/latest/dg/t_Loading_tables_with_the_COPY_command.html

File: test/metabase/api/dataset_test.clj (hunk 2 above)
Comment (robdaemon) [reply test/metabase/api/dataset_test.clj:53]: Yeah, Oracle would bite us here if our cluster were a larger RAC cluster. Oracle does not guarantee ordering of results (due to parallelism in RAC) unless an `ORDER BY` is used.

**Review by robdaemon — APPROVED** [review]

**Pull request merged**